test/functional/test_solution.py

```diff
@@ -0,0 +1,439 @@
+            'tenant_username':          'testlab',
+            'tenant_password':          'changeme',
+            'client_subnet':            'client-v4-subnet',
+            'guest_username':           'centos',
+            'guest_password':           'changeme',
+            'server_http_port':         '8080',
+            'server_client_ip':         '10.2.2.3'
+        }
+        self.symbols.update(new_symbols)
+
+
+def nclientmanager(symbols):
+    nclient_config = {
+        'username': symbols['tenant_username'],
+        'password': symbols['tenant_password'],
+        'tenant_name': symbols['tenant_name'],
+        'auth_url': symbols['openstack_auth_url']
+    }
+
+    neutronclient = client.Client(**nclient_config)
+    return NeutronClientPollingManager(neutronclient)
+
+
+class LBaaSv1(object):
+    def __init__(self, symbols):
+        self.polling_interval = 1
+        self.symbols = symbols
+        self.ncm = nclientmanager(self.symbols)
+        self.proxies = []
+
+    def clear_pools(self):
+        for pool in self.ncm.list_pools()['pools']:
+            self.ncm.delete_pool(pool['id'])
+        while self.ncm.list_pools()['pools']:
+            time.sleep(1)
+
+    def create_pool(self):
+        conf = {}
+        for sn in self.ncm.list_subnets()['subnets']:
+            if sn['name'] == self.symbols['client_subnet']:
+                conf = {'tenant_id':  sn['tenant_id'],
+                        'lb_method':  'ROUND_ROBIN',
+                        'protocol':   'HTTP',
+                        'subnet_id':  sn['id'],
+                        'provider':   self.symbols['lb_provider'],
+                        'name':       'test_pool_01'}
+
+        return self.ncm.create_pool({'pool': conf})['pool']
+
+    def wait_for_object_state(self, field, value, method, key, *args):
+        time.sleep(self.polling_interval)
+        current_state = method(*args)[key]
+        current_value = current_state[field]
+        while value != current_value:
+            sys.stdout.flush()
+            current_state = method(*args)[key]
+            current_value = current_state[field]
+            time.sleep(self.polling_interval)
+
+    def create_vip(self, pool_id, subnet_id):
+        conf = {
+            'name': 'test_vip_01',
+            'pool_id': pool_id,
+            'subnet_id': subnet_id,
+            'protocol': 'HTTP',
+            'protocol_port': 80
+        }
+        return self.ncm.create_vip({'vip': conf})['vip']
+
+    def create_member(self, pool_id):
+        conf = {
+            'pool_id': pool_id,
+            'address': self.symbols['server_client_ip'],
+            'protocol_port': int(self.symbols['server_http_port'])
+        }
+        return self.ncm.create_member({'member': conf})['member']
+
+    def create_healthmonitor(self):
+        conf = {
+            'type': 'HTTP',
+            'delay': 10,
+            'timeout': 3,
+            'max_retries': 2
+        }
```

Comment:
Same comments, the config is static and we probably want to be able to change these for people to reuse the code.